### ISSUE TITLE
[health 8.0.0] There was an error getting the workout data!

### ISSUE THREAD

**Issue by debiotech4ayo** [cph-cachet/flutter-plugins#791]
### Device / Emulator and OS

Physical devices :

 - Device: Power U 10
 - OS: Android 11

And

- Device: Oppo Reno 4 pro
- OS: Android 12

### Describe the bug
I can't access the Google Fit workout data.

### To Reproduce
I cloned the project, opened the example project and used the types defined in lines 40 instead of line 38. Run the application, doing the authentication and clicked on "fetch data".

### Expected behavior
I expected to be able to fetch the activities I manually added in google fit. Instead, I'm only able to fetch steps count.

### Actual behavior
```
W/FLUTTER_HEALTH::ERROR( 3755): There was an error getting the workout data!
W/FLUTTER_HEALTH::ERROR( 3755): 4: The user must be signed in to make this API call.
W/FLUTTER_HEALTH::ERROR( 3755): [Ljava.lang.StackTraceElement;@79ff3d9
W/FLUTTER_HEALTH::ERROR( 3755): There was an error getting the data!
W/FLUTTER_HEALTH::ERROR( 3755): 4: The user must be signed in to make this API call.
W/FLUTTER_HEALTH::ERROR( 3755): [Ljava.lang.StackTraceElement;@20e099b
W/FLUTTER_HEALTH::ERROR( 3755): There was an error getting the data!
W/FLUTTER_HEALTH::ERROR( 3755): 4: The user must be signed in to make this API call.
W/FLUTTER_HEALTH::ERROR( 3755): [Ljava.lang.StackTraceElement;@13178e4
I/flutter ( 3755): HealthDataPoint -
I/flutter ( 3755):     value: 11,
I/flutter ( 3755):     unit: COUNT,
I/flutter ( 3755):     dateFrom: 2023-08-21 18:39:30.993,
I/flutter ( 3755):     dateTo: 2023-08-21 18:39:54.134,
I/flutter ( 3755):     dataType: STEPS,
I/flutter ( 3755):     platform: PlatformType.ANDROID,
I/flutter ( 3755):     deviceId: RP1A.200720.011,
I/flutter ( 3755):     sourceId: raw:com.google.step_count.cumulative:OPPO:CPH2089:125a4426:pedometer  Non-wakeup,
I/flutter ( 3755):     sourceName: CPH2089
```

### Flutter doctor 
```
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel stable, 3.10.6, on Microsoft Windows [Version 10.0.22621.2134], locale fr-CH)
[√] Windows Version (Installed version of Windows is version 10 or higher)
[√] Android toolchain - develop for Android devices (Android SDK version 34.0.0-rc4)
[√] Chrome - develop for the web
[√] Visual Studio - develop for Windows (Visual Studio Community 2022 17.5.3)
[√] Android Studio (version 2022.1)
[√] VS Code (version 1.81.1)
[√] Connected device (4 available)
[√] Network resources                

• No issues found!
```

### Additional information
The SHA-1 key is working as I'm able to get steps count. I tried to add all scopes relatives to Fitness API without success.

**Comment by debiotech4ayo** [comment]
What did I do :
- Testing version 7.0.1, 7.0.0, 6.0.0, 4.6.0 -> no one can fetch workout data
- [Reading this issue](https://github.com/cph-cachet/flutter-plugins/issues/775) I've added all permissions to version 4.6.0 without success
- In google cloud, I created a new project with the app package name, added my email adress as tester, added all Fitness API scopes, added the OAuth SHA-1 key.
- Installed [Package manager](https://f-droid.org/en/packages/com.smartpack.packagemanager/) to verify SHA-1 key.

With version 4.6.0, I'm able to view steps count, to add for example blood pressure and to delete data. But I'm not able to fetch workout data or add workout data, with always the same error "The user must be signed in to make this API call.". 

I'm logging in with the same email adress as added in google cloud.

As I have the same problems with several versions of the application, I'm guessing I'm doing something wrong somewhere and it's not necessarily app problem.

**Comment by debiotech4ayo** [comment]
Ok I found a workaround : I added "HealthDataType.DISTANCE_DELTA," along with "HealthDataType.WORKOUT," to the types list as explained in this [issue](https://github.com/cph-cachet/flutter-plugins/issues/613). 

Then the dialog asking for access rights for distance and workout will pop up. If the user accept it, it works. When using WORKOUT alone, the dialog for access rights doesn't pop up.

I don't need to add additionnal permissions as I'm using Google Fit. I only have these :
```
    <uses-permission android:name="android.permission.ACTIVITY_RECOGNITION"/>
    <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION"/>
    <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION"/>
```
